sstables/mx/reader.cc

```diff
@@ -1591,8 +1590,8 @@ class mx_sstable_mutation_reader : public mp_row_consumer_reader_mx {
             on_internal_error(sstlog, "mx reader: fast_forward_to(partition_range) not supported for reversed queries");
         }
 
-        return ensure_initialized().then([this, &pr] {
-            if (!is_initialized()) {
+        return maybe_initialize().then([this, &pr] (bool initialized) {
+            if (!initialized) {
```

Comment:
(if we get a nullopt we return an empty_reader rather than an sstable_reader that was primed to look empty)